File: apps/backend/src/main/java/com/archaeologist/domain/model/AgentResult.java
package com.archaeologist.domain.model;

import java.time.LocalDateTime;
import java.util.UUID;

public record AgentResult(
    UUID id,
    UUID jobId,
    String agentName,
    AgentStatus status,
    String output,
    String errorMessage,
    LocalDateTime startedAt,
    LocalDateTime completedAt,
    int executionOrder
) {}

File: apps/analyzer/src/domain/models/agent_result.py
"""Agent output/result domain models."""

from dataclasses import dataclass, field
from datetime import datetime
from enum import Enum
from typing import Optional


class AgentStatus(str, Enum):
    PENDING = "pending"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"
    SKIPPED = "skipped"


@dataclass
class AgentResult:
    """
    Captures the outcome of a single agent execution within a pipeline run.

    Stored in the ``agent_results`` table; one row per agent per job.
    """

    agent_name: str = ""
    status: AgentStatus = AgentStatus.PENDING
    output: Optional[dict] = None
    error_message: Optional[str] = None
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    execution_order: int = 0
